protocol/communities_events_utils_test.go

```diff
@@ -363,9 +364,20 @@ func setUpOnRequestCommunityAndRoles(base CommunityEventsTestsInterface, role pr
 	checkPermissionGranted := func(response *MessengerResponse) error {
 		return checkRolePermissionInResponse(response, base.GetEventSender().IdentityPublicKey(), role)
 	}
-
 	waitOnMessengerResponse(s, WaitCommunityCondition, checkPermissionGranted, base.GetMember())
 
+	for _, eventSender := range additionalEventSenders {
+		advertiseCommunityTo(s, community, base.GetControlNode(), eventSender)
+		joinOnRequestCommunity(s, community, base.GetControlNode(), eventSender)
+
+		grantPermission(s, community, base.GetControlNode(), eventSender, role)
+		checkPermissionGranted = func(response *MessengerResponse) error {
+			return checkRolePermissionInResponse(response, eventSender.IdentityPublicKey(), role)
+		}
+		waitOnMessengerResponse(s, WaitCommunityCondition, checkPermissionGranted, base.GetMember())
+		time.Sleep(500 * time.Millisecond)
```

Comment:
this should work without sleep

Reply:
Ah, this has been flaky so I've added the sleep here, but didn't mean to leave it in.
I'm still making changes to this PR.

Any other ideas what can be done to make this less flaky?

Reply:
0. Need to detect where it is failing exactly
1. WaitCommunityCondition change on condition, where you will wait for a community + specific permission for a specific user
2. Call waitOnMessengerResponse for all other members in order for they have the latest CommunityDescription

P.S. If there will be any problems, ping me, I'll take a look at logs